api/src/shared/infrastructure/jobs/JobPgBoss.js

```diff
@@ -1,27 +1,44 @@
+import { DomainTransaction } from '../../domain/DomainTransaction.js';
+
 class JobPgBoss {
-  constructor(config, queryBuilder) {
+  constructor(config) {
     this.name = config.name;
     this.retryLimit = config.retryLimit ?? 0;
     this.retryDelay = config.retryDelay ?? 30;
     this.retryBackoff = config.retryBackoff || false;
     this.expireIn = config.expireIn || '00:15:00';
-    this.queryBuilder = queryBuilder;
   }
 
-  async schedule(data) {
-    await this.queryBuilder('pgboss.job').insert({
+  #buildPayload(data) {
+    return {
       name: this.name,
       retrylimit: this.retryLimit,
       retrydelay: this.retryDelay,
       retrybackoff: this.retryBackoff,
       expirein: this.expireIn,
       data,
       on_complete: true,
-    });
+    };
+  }
+
+  async #schedule(jobs) {
+    const knexConn = DomainTransaction.getConnection();
+
+    return knexConn('pgboss.job').insert(jobs);
   }
 
   async performAsync(data) {
-    return this.schedule(data);
+    const job = this.#buildPayload(data);
+
+    return this.#schedule(job);
+  }
+
+  async performMultipleJobsAsync(datas) {
+    const jobs = datas.map((data) => {
+      return this.#buildPayload(data);
+    });
+
+    return this.#schedule(jobs);
```

Comment:
Je propose l simplification suivante pour n'avoir qu'une fonction et etre + DRY

```suggestion
async performAsync(...datas) {
  const jobs = datas.map((data) => {
     return this.#buildPayload(data);
   });

    return this.#schedule(jobs);
  }
```

Ce qui fait que les deux versions fonctionnent 

```
await performAsync('a');
await performAsync('b', 'c');
```

Reply:
```js
const jobDatas = ['a', 'b']
await performAsync(...jobDatas);
```